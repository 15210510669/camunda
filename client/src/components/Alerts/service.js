import {get, post, del, put} from 'request';

export async function loadAlerts() {
  const response = await get('/api/alert');

  return await response.json();
}

export async function loadReports() {
  const response = await get('/api/report');

  return await response.json();
}

export async function saveNewAlert(alert) {
  const response = await post('/api/alert', alert);

  const json = await response.json();

  return json.id;
}

export async function updateAlert(id, alert) {
  return await put('/api/alert/' + id, alert);
}

export async function deleteAlert(id) {
  return await del('api/alert/' + id);
}

