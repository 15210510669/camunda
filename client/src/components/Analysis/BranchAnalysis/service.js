/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {post} from 'request';

export async function loadFrequencyData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter
) {
  const response = await post(
    'api/report/evaluate',
    createFlowNodeFrequencyReport(
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      filter
    )
  );

  return await response.json();
}

function createFlowNodeFrequencyReport(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter
) {
  return {
    combined: false,
    reportType: 'process',
    data: {
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      filter,
      view: {
        entity: 'flowNode',
        property: 'frequency',
      },
      groupBy: {
        type: 'flowNodes',
        unit: null,
      },
      visualization: 'heat',
    },
  };
}

export async function loadCorrelationData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter,
  gateway,
  end
) {
  const response = await post('api/analysis/correlation', {
    processDefinitionKey,
    processDefinitionVersions,
    tenantIds,
    filter,
    gateway,
    end,
  });

  return await response.json();
}
