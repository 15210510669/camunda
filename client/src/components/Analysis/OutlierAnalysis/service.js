/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get} from 'request';

export async function loadNodesOutliers(config) {
  const response = await get('api/analysis/flowNodeOutliers', config);
  return await response.json();
}

export {loadDurationData, loadCommonOutliersVariables, getInstancesDownloadUrl} from './service.ts';
