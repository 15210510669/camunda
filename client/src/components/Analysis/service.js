import {get, post} from 'request';

export async function loadProcessDefinitionXml(processDefinitionKey, processDefinitionVersion) {
  const response = await get('/api/process-definition/xml', {processDefinitionKey, processDefinitionVersion});

  return await response.text();
}

export async function loadFrequencyData(processDefinitionKey, processDefinitionVersion, filter) {
  const response = await post('/api/process-definition/heatmap/frequency', {
    processDefinitionKey,
    processDefinitionVersion,
    filter
  });

  return await response.json();
}

export async function loadCorrelationData(processDefinitionKey, processDefinitionVersion, filter, gateway, end) {
  const response = await post('/api/process-definition/correlation', {
    processDefinitionKey,
    processDefinitionVersion,
    filter,
    gateway,
    end
  });

  return await response.json();
}
