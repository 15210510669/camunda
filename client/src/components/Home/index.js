/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {default as Home} from './Home';
export {default as Collection} from './Collection';
