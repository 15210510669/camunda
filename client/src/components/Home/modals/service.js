/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {get} from 'request';
import {t} from 'translation';

export async function searchIdentities(terms) {
  const response = await get(`api/identity/search`, {terms});

  return await response.json();
}

export async function getDefinitionTenants(defintionKey, defintionType) {
  const response = await get(`api/definition/${defintionType}/${defintionKey}`);

  return await response.json();
}

export async function getDefinitionsWithTenants() {
  const response = await get('api/definition');

  return await response.json();
}

export async function getTenantsWithDefinitions() {
  const response = await get('api/definition/_groupByTenant');

  return await response.json();
}

export function formatDefintionName({name, type}) {
  const typeLabel = type === 'process' ? t('home.sources.process') : t('home.sources.decision');

  return name + ' (' + typeLabel + ')';
}
