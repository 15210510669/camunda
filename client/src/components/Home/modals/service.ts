/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get} from 'request';
import {Definition, Tenant} from 'types';

export type TenantWithDefinitions = Tenant & {definitions: Definition};

export type DefinitionWithTenants = Omit<Definition, 'key'> & {tenants: Tenant[]; key: string};

export async function getDefinitionsWithTenants(): Promise<DefinitionWithTenants[]> {
  const response = await get('api/definition');

  return await response.json();
}

export async function getTenantsWithDefinitions(): Promise<TenantWithDefinitions[]> {
  const response = await get('api/definition/_groupByTenant');

  return await response.json();
}
