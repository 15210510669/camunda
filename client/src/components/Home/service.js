/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {get, post, put, del} from 'request';

export async function loadEntities() {
  const response = await get('api/entities');
  return await response.json();
}

export async function addUser(collection, id, type, role) {
  return await post(`api/collection/${collection}/role`, {identity: {id, type}, role});
}

export async function editUser(collection, id, role) {
  return await put(`api/collection/${collection}/role/${id}`, {role});
}

export async function removeUser(collection, id) {
  return await del(`api/collection/${collection}/role/${id}`);
}

export async function loadAlerts(collection) {
  const response = await get(`api/collection/${collection}/alerts`);
  return await response.json();
}

export async function addAlert(alert) {
  return await post(`api/alert`, alert);
}

export async function editAlert(id, alert) {
  return await put(`api/alert/${id}`, alert);
}

export async function removeAlert(id) {
  return await del(`api/alert/${id}`);
}

export async function copyEntity(type, id, name, collectionId) {
  const query = {name};

  if (collectionId || collectionId === null) {
    query.collectionId = collectionId;
  }

  const response = await post(`api/${type}/${id}/copy`, undefined, {query});
  const json = await response.json();

  return json.id;
}

export async function searchIdentities(terms) {
  const response = await get(`api/identity/search`, {terms});

  return await response.json();
}
