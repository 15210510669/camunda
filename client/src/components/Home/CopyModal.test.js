/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {Switch} from 'components';

import CopyModalWithErrorHandling from './CopyModal';

import {loadEntities} from './service';

const CopyModal = CopyModalWithErrorHandling.WrappedComponent;

jest.mock('./service', () => ({
  loadEntities: jest.fn().mockReturnValue([
    {
      id: 'aCollectionId',
      name: 'aCollectionName',
      lastModified: '2017-11-11T11:11:11.1111',
      created: '2017-11-11T11:11:11.1111',
      owner: 'user_id',
      lastModifier: 'user_id',
      entityType: 'collection',
      currentUserRole: 'manager',
      data: {
        subEntityCounts: {
          dashboard: 2,
          report: 8
        },
        roleCounts: {
          user: 5,
          group: 2
        }
      }
    },
    {
      id: 'aDashboardId',
      name: 'aDashboard',
      lastModified: '2017-11-11T11:11:11.1111',
      created: '2017-11-11T11:11:11.1111',
      owner: 'user_id',
      lastModifier: 'user_id',
      entityType: 'dashboard',
      currentUserRole: 'editor',
      data: {
        subEntityCounts: {
          report: 8
        },
        roleCounts: {}
      }
    },
    {
      id: 'aReportId',
      name: 'aReport',
      lastModified: '2017-11-11T11:11:11.1111',
      created: '2017-11-11T11:11:11.1111',
      owner: 'user_id',
      currentUserRole: 'editor',
      data: {subEntityCounts: {}, roleCounts: {}},
      lastModifier: 'user_id',
      reportType: 'process', // or 'decision'
      combined: false,
      entityType: 'report'
    }
  ])
}));

const props = {
  mightFail: jest.fn().mockImplementation((data, cb) => cb(data)),
  entity: {name: 'Test Dashboard', entityType: 'dashboard', data: {subEntityCounts: {report: 2}}},
  collection: 'aCollectionId',
  onConfirm: jest.fn(),
  onClose: jest.fn()
};

it('should match snapshot', () => {
  const node = shallow(<CopyModal {...props} />);

  expect(node).toMatchSnapshot();
});

it('should load available collections', () => {
  shallow(<CopyModal {...props} />);

  expect(loadEntities).toHaveBeenCalled();
});

it('should offer to move the copy', () => {
  const node = shallow(<CopyModal {...props} jumpToEntity />);

  node.find(Switch).simulate('change', {target: {checked: true}});

  expect(node).toMatchSnapshot();
});

it('should hide option to move the copy for collection entities', () => {
  const node = shallow(
    <CopyModal {...props} entity={{name: 'collection', entityType: 'collection'}} />
  );

  expect(node).toMatchSnapshot();
});

it('should call the onConfirm action', () => {
  const node = shallow(<CopyModal {...props} jumpToEntity />);

  node.find('.confirm').simulate('click');

  expect(props.onConfirm).toHaveBeenCalledWith('Test Dashboard (copy)', true, false);
});

it('should hide the jump checkbox if jumpToEntity property is not added', () => {
  const node = shallow(<CopyModal {...props} />);

  expect(node.find('LabeledInput[type="checkbox"]')).not.toExist();

  node.find('.confirm').simulate('click');

  expect(props.onConfirm).toHaveBeenCalledWith('Test Dashboard (copy)', undefined, false);
});
