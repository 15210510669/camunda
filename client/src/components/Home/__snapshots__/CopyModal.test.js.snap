// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should hide option to move the copy for collection entities 1`] = `
<Modal
  className="CopyModal"
  onClose={[MockFunction]}
  onConfirm={[Function]}
  open={true}
>
  <Component>
    Copy collection
  </Component>
  <Component>
    <Form>
      <FormGroup>
        <LabeledInput
          autoComplete="off"
          label="Name of Copy"
          onChange={[Function]}
          type="text"
          value="collection (copy)"
        />
      </FormGroup>
      <span>
        All contained items will be copied, and all assigned users will be added to the new collection, too.
      </span>
    </Form>
  </Component>
  <Component>
    <ForwardRef(Button)
      className="cancel"
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      className="confirm"
      color="blue"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Copy
    </ForwardRef(Button)>
  </Component>
</Modal>
`;

exports[`should match snapshot 1`] = `
<Modal
  className="CopyModal"
  onClose={[MockFunction]}
  onConfirm={[Function]}
  open={true}
>
  <Component>
    Copy Test Dashboard
  </Component>
  <Component>
    <Form>
      <FormGroup>
        <LabeledInput
          autoComplete="off"
          label="Name of Copy"
          onChange={[Function]}
          type="text"
          value="Test Dashboard (copy)"
        />
      </FormGroup>
      <FormGroup
        className="moveSection"
      >
        <Switch
          checked={false}
          label="Move copy to …"
          onChange={[Function]}
        />
      </FormGroup>
    </Form>
  </Component>
  <Component>
    <ForwardRef(Button)
      className="cancel"
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      className="confirm"
      color="blue"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Copy
    </ForwardRef(Button)>
  </Component>
</Modal>
`;

exports[`should offer to move the copy 1`] = `
<Modal
  className="CopyModal"
  onClose={[MockFunction]}
  onConfirm={[Function]}
  open={true}
>
  <Component>
    Copy Test Dashboard
  </Component>
  <Component>
    <Form>
      <FormGroup>
        <LabeledInput
          autoComplete="off"
          label="Name of Copy"
          onChange={[Function]}
          type="text"
          value="Test Dashboard (copy)"
        />
      </FormGroup>
      <FormGroup
        className="moveSection"
      >
        <Switch
          checked={true}
          label="Move copy to …"
          onChange={[Function]}
        />
      </FormGroup>
      <FormGroup
        noSpacing={true}
      >
        <Typeahead
          formatter={[Function]}
          initialValue={null}
          noValuesMessage="No collections have been created"
          onSelect={[Function]}
          placeholder="Please select"
          values={
            Array [
              Object {
                "entityType": "collection",
                "id": null,
                "name": "Home",
              },
            ]
          }
        />
      </FormGroup>
      <FormGroup>
        This Dashboard contains 2 Reports. They will be copied, too.
      </FormGroup>
      <FormGroup>
        <LabeledInput
          checked={true}
          label="Go to new location"
          onChange={[Function]}
          type="checkbox"
        />
      </FormGroup>
    </Form>
  </Component>
  <Component>
    <ForwardRef(Button)
      className="cancel"
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Button)>
    <ForwardRef(Button)
      className="confirm"
      color="blue"
      disabled={true}
      onClick={[Function]}
      variant="primary"
    >
      Copy
    </ForwardRef(Button)>
  </Component>
</Modal>
`;
